src/mo_types/type.ml

```diff
@@ -1049,6 +1059,11 @@ let rec rel_typ d rel eq t1 t2 =
   | _, _ -> false
   end
 
+and rel_sort s1 s2 =
+  match s1, s2 with
+  | Local Stable, Local Flexible -> true (* stable function can be assigned to flexible function *)
```

Comment:
This should probably require equality when checking is invariant (rel == eq)

Reply:
I changed this here to consider the generic type parameters.
Stable functions with stable type parameters cannot be promoted to flexible functions with implicitly flexible type parameters.

I thought `rel` would always be sub-typing but probably I missed a `RelArg` on equality?